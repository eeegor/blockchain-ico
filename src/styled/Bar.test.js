import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Bar } from '.';

describe('Bar', () => {
	it('renders horizontal bar', () => {
		const tree = renderer.create(<Bar />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('display', 'flex');
	});

	it('renders horizontal bar with centered content', () => {
		const tree = renderer.create(<Bar center />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('align-items', 'center');
		expect(tree).toHaveStyleRule('justify-content', 'center');
	});

	it('renders horizontal bar with horizontally centered content', () => {
		const tree = renderer.create(<Bar centerX />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('justify-content', 'center');
	});

	it('renders horizontal bar with horizontally distibuted content', () => {
		const tree = renderer.create(<Bar justifyBetween />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('justify-content', 'space-between');
	});

	it('renders horizontal bar with vertically centered content', () => {
		const tree = renderer.create(<Bar centerY />).toJSON();
		expect(tree).toMatchSnapshot();
		expect(tree).toHaveStyleRule('align-items', 'center');
	});
});
