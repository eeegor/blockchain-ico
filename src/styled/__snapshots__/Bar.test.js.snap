// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar renders horizontal bar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div
  className="c0"
/>
`;

exports[`Bar renders horizontal bar with centered content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`Bar renders horizontal bar with horizontally centered content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`Bar renders horizontal bar with horizontally distibuted content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
/>
`;

exports[`Bar renders horizontal bar with vertically centered content 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0"
/>
`;
